gen-client.sh

```diff
@@ -61,15 +61,23 @@ else
   VOLUME_DIR="${PWD}"
 fi
 
+REMOVE_COOKIE_AUTH_FILTER='del(.paths[][].security|select(.)[]|select(.cookieAuth))|del(.components.securitySchemes.cookieAuth)'
+
+# These two may be needed when upgrading the generator image
+FIX_TASK_CREATED_RESOURCES_FILTER='(.components.schemas.TaskResponse|select(.)|.properties.created_resources.items) |= {"$oneOf":[{type:"null"},.]}'
+FIX_TASK_ERROR_FILTER='(.components.schemas.TaskResponse|select(.)|.properties.error) |= (del(.readOnly) | .additionalProperties.type = "string")'
```

Comment:
Can you explain what these filters are doing?

Reply:
Both of them are necessary, because the newer templates do more sophisticated validation on the http responses than before.
The first one is to allow the created_resources list to contain `null` entries, which happens when we delete a resource formerly locked by the task. (It is not even evident to me whether it's better to not report them on the server side anymore, but as a matter of fact, current pulp violates the schema here.)
The second one, I'm not even sure this is the right thing to do. But what is does: It removes the `readOnly` property from the `error` attribute and changes its `additionalProperties` to be strings. I encountered this in some tests, where the bindings were unable to instantiate a `TaskResponse` object for the `readOnly` thing and it's unclear to me why this should be an issue.

Reply:
I wrote these filters as Variables (Constants) because they may be useful to the other generators too. Who knows...